src/vcpkg/base/system.process.cpp

```diff
@@ -587,6 +587,7 @@ namespace vcpkg
             // Enables proxy information to be passed to Curl, the underlying download library in cmake.exe
             "http_proxy",
             "https_proxy",
+            "no_proxy",
```

Comment:
This part of the change seems good. Handling for this should probably be added below the 

    auto maybe_https_proxy_env = get_environment_variable(EnvironmentVariableHttpsProxy);

to make the WinHTTP backend understand `NO_PROXY` rather than trying to engage WPAD.